include/abb_librws/v1_0/rws_client.h

```diff
@@ -155,6 +156,18 @@ class RWSClient
                                        const std::string& tool = "",
                                        const std::string& wobj = "");
 
+
+  /**
+   * \brief Checks progress of provided resource.
+   * 
+   * \param id Resource id to check
+   *
+   * \return Returns progress of resource.
+   *
+   * \throw \a RWSError if something goes wrong.
+   */
+  progress::ProgressInfo getProgress(std::string const& id);
```

Comment:
This function should not be a member of `RWSClient`. The idea is that `RWSClient` is simply the communication layer, and the RWS-specific logic is implemented in free functions. The member functions like `getMechanicalUnitRobTarget()` that you see are rudimentary and should be converted to free functions or removed.